packages/editor/src/components/block-settings-menu/content-only-settings-menu.js

```diff
@@ -19,53 +20,81 @@ import { store as editorStore } from '../../store';
 import { unlock } from '../../lock-unlock';
 
 function ContentOnlySettingsMenuItems( { clientId, onClose } ) {
-	const { entity, onNavigateToEntityRecord } = useSelect(
-		( select ) => {
-			const {
-				getBlockEditingMode,
-				getBlockParentsByBlockName,
-				getSettings,
-				getBlockAttributes,
-			} = select( blockEditorStore );
-			const contentOnly =
-				getBlockEditingMode( clientId ) === 'contentOnly';
-			if ( ! contentOnly ) {
-				return {};
-			}
-			const patternParent = getBlockParentsByBlockName(
-				clientId,
-				'core/block',
-				true
-			)[ 0 ];
+	const { entity, onNavigateToEntityRecord, canEditTemplateParts } =
+		useSelect(
+			( select ) => {
+				const {
+					getBlockEditingMode,
+					getBlockParentsByBlockName,
+					getSettings,
+					getBlockAttributes,
+				} = select( blockEditorStore );
+				const contentOnly =
+					getBlockEditingMode( clientId ) === 'contentOnly';
+				if ( ! contentOnly ) {
+					return {};
+				}
+				const patternParent = getBlockParentsByBlockName(
+					clientId,
+					'core/block',
+					true
+				)[ 0 ];
 
-			let record;
-			if ( patternParent ) {
-				record = select( coreStore ).getEntityRecord(
-					'postType',
-					'wp_block',
-					getBlockAttributes( patternParent ).ref
+				let record;
+				const _canEditTemplateParts = select( coreStore ).canUser(
+					'create',
+					'templates'
 				);
-			} else {
-				const { getCurrentPostType, getCurrentTemplateId } =
-					select( editorStore );
-				const currentPostType = getCurrentPostType();
-				const templateId = getCurrentTemplateId();
-				if ( currentPostType === 'page' && templateId ) {
+				if ( patternParent ) {
 					record = select( coreStore ).getEntityRecord(
 						'postType',
-						'wp_template',
-						templateId
+						'wp_block',
+						getBlockAttributes( patternParent ).ref
 					);
+				} else {
+					const { getCurrentTemplateId } = select( editorStore );
+					const templateId = getCurrentTemplateId();
+					if ( templateId && _canEditTemplateParts ) {
```

Comment:
Probably don't need `&& _canEditTemplateParts` here given the `if ( ! canEditTemplateParts ) {` below.